ipatests/test_cmdline/test_cli.py

```diff
@@ -317,7 +318,7 @@ def test_without_options():
     def test_certfind(self):
         with tempfile.NamedTemporaryFile() as f:
             f.write(goodcert_headers)
-            f.flush()
+            ipautil.flush_sync(f)
```

Comment:
You don't have to add flush to all places, only when it's a critical permanent file. This code and other places in ``ipatests`` only create temporary files, that will be deleted any way. ``fsync`` is a costly operation, because it forces the Kernel and disk driver to bypass write caches.

Reply:
This is also true outside `ipatests`, right? In various places a `NamedTemporaryFile` is created only to be used as an argument for some outside tool (e.g. `openssl verify -CAfile {temp_file}) and deleted after that.

Reply:
Excellent analysis. Yes, it's only relevant for persistent files and not for places that use a short-lived temporary file. We mostly care about config files, certs, keys, passwords etc.